temporal-normalizer/src/main/java/edu/illinois/cs/cogcomp/temporal/normalizer/main/TemporalChunkerAnnotator.java

```diff
@@ -0,0 +1,319 @@
+                    && clabel != null) {
+
+                if (previous != null) {
+                    currentChunkEnd = previous.getEndSpan();
+                    Constituent temp_label =
+                            new Constituent(clabel, ViewNames.TIMEX3, record,
+                                    currentChunkStart, currentChunkEnd);
+                    try {
+                        clabel = heidelTimeNormalize(temp_label);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                    Constituent label =
+                            new Constituent(clabel, ViewNames.TIMEX3, record,
+                                    currentChunkStart, currentChunkEnd);
+                    chunkView.addConstituent(label);
+                    clabel = null;
+                } // else no chunk in progress (we are at the start of the doc)
+            }
+
+            if (lbjtoken.type.charAt(0) == 'B') {
+                currentChunkStart = current.getStartSpan();
+                clabel = lbjtoken.type.substring(2);
+            }
+            previous = current;
+            tcounter++;
+        }
+        if (clabel != null && null != previous) {
+            currentChunkEnd = previous.getEndSpan();
+            Constituent temp_label =
+                    new Constituent(clabel, ViewNames.TIMEX3, record,
+                            currentChunkStart, currentChunkEnd);
+            try {
+                clabel = heidelTimeNormalize(temp_label);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            Constituent label =
+                    new Constituent(clabel, ViewNames.TIMEX3, record,
+                            currentChunkStart, currentChunkEnd);
+            chunkView.addConstituent(label);
+        }
+        record.addView(ViewNames.TIMEX3, chunkView);
+
+        return; // chunkView;
+    }
+
+    /**
+     * Use this function to add specific document creation time
+     * The default DCT is current date
+     * @param date the DCT you want to set
+     */
+    public void addDocumentCreationTime(String date) {
+        SimpleDateFormat f = new SimpleDateFormat("dd-MMM-yyyy");
+        try {
+            this.dct = f.parse(date);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Normalize temporal phrase
+     * @param temporal_phrase
+     * @return
+     * @throws Exception
+     */
+    private String heidelTimeNormalize(Constituent temporal_phrase) throws Exception {
+        // If user didn't specify document creation date, use the current date
+        if (this.dct == null) {
+            this.dct = new Date();
+        }
+
+        String xml_res = this.heidelTime.process(temporal_phrase.toString(), this.dct);
+
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
Shouldn't this be a field, not a local variable? It looks like it will get instantiated every time you process a temporal phrase.